app/security/Roles.scala

```diff
@@ -347,6 +347,8 @@ object RoleUtils {
 
   def isSdipFaststream(implicit user: CachedData): Boolean = user.application exists (_.applicationRoute == ApplicationRoute.SdipFaststream)
 
+  def isSdipFaststreamFailed(implicit user: CachedData): Boolean = user.application.fold(false)(_.progress.sdipFSFailed)
```

Comment:
I think it depends... personally I find fold more intuitive than forAll. What about a getOrElse?